src/Symfony/Component/HttpClient/RetryHttpClient.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpClient\Response\AsyncContext;
+use Symfony\Component\HttpClient\Response\AsyncResponse;
+use Symfony\Component\HttpClient\Retry\RetryStrategyInterface;
+use Symfony\Contracts\HttpClient\ChunkInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * Automatically retries the failing requests.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class RetryHttpClient implements HttpClientInterface
+{
+    use AsyncDecoratorTrait;
+
+    private $strategy;
+    private $logger;
+
+    private $retryCount = 0;
+
+    public function __construct(HttpClientInterface $client, RetryStrategyInterface $strategy, LoggerInterface $logger = null)
+    {
+        $this->client = $client;
+        $this->strategy = $strategy;
+        $this->logger = $logger ?: new NullLogger();
+    }
+
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        return new AsyncResponse($this->client, $method, $url, $options, function (ChunkInterface $chunk, AsyncContext $context) use ($method, $url, $options) {
+            $exception = null;
+            try {
+                if ($chunk->isTimeout() || null !== $chunk->getInformationalStatus()) {
+                    yield $chunk;
+
+                    return;
+                }
+
+                // only retry first chunk
+                if (!$chunk->isFirst()) {
+                    $context->passthru();
+                    yield $chunk;
+
+                    return;
+                }
+            } catch (TransportExceptionInterface $exception) {
+                // catch TransportExceptionInterface to send it to strategy.
+            }
+
+            if (!$this->strategy->isRetryable($this->retryCount, $response = $context->getResponse(), $exception)) {
+                $context->passthru();
+                yield $chunk;
+
+                return;
+            }
+
+            ++$this->retryCount;
+
+            $context->setInfo('retry', $this->retryCount);
+            $response->cancel();
+
+            $delay = $this->strategy->getWaitingTime($this->retryCount, $response, $exception);
+
+            $this->logger->info('Error returned by the server. Retrying #{retryCount} using {delay} ms delay. Error: "{error}"', [
+                'retryCount' => $this->retryCount,
+                'delay' => $delay,
+                'error' => ($exception ? $exception->getMessage() : 'StatusCode: '.$response->getStatusCode()),
```

Comment:
If `$exception === null`, this will read "Error: StatusCode: 200"

I suggest moving "Error:" from row 81

Reply:
removed `error:`